yamux/src/session.rs

```diff
@@ -1094,4 +1115,30 @@ mod test {
             assert_eq!(reset_msg.stream_id(), 5)
         });
     }
+
+    #[test]
+    fn test_remote_does_not_respond_go_away() {
+        let mut rt = tokio::runtime::Runtime::new().unwrap();
+
+        rt.block_on(async {
+            let (_remote, local) = MockSocket::new();
+            let mut config = Config::default();
+            config.enable_keepalive = false;
+            config.connection_write_timeout = Duration::from_secs(1);
+
+            let mut session = Session::new_server(local, config);
+
+            let mut control = session.control();
+            tokio::spawn(async move {
+                let _ignore = control.close().await;
+            });
+
+            while let Some(Ok(mut stream)) = session.next().await {
```

Comment:
Since this test has no assertions, it's better to add some comments about what's expected here.